Tasks/Common/npm-common/npmregistry.ts

```diff
@@ -95,6 +89,32 @@ export class NpmRegistry implements INpmRegistry {
         return new NpmRegistry(url, auth, authOnly);
     }
 
+    // make a request to the endpoint uri, and take a look at the response header to
+    // determine whether this is our service, or an external service.
+    private static async isEndpointInternal(endpointUri: string): Promise<boolean> {
+        const proxyUrl: string = tl.getVariable('agent.proxyurl');
```

Comment:
I am glad you remembered to do this; however, rather than rolling our own logic to determine whether or not we need to use a proxy.  You should use tasklib's [getHttpProxyConfiguration](https://github.com/Microsoft/vsts-task-lib/blob/master/node/task.ts#L1598).  It will handle proxy bypassing for you (which you aren't handling here).